src/managers/config.py

```diff
@@ -102,11 +104,13 @@ class ConfigManager:
 
     def __init__(
         self,
+        role: Role,
```

Comment:
**thought, issue** I wouldn't honestly overload this class with both `kafka broker` AND `cruise control` configuration logic. I believe it is sane to have two different classes: one that handles kakfa configuration and another one that handle cruise-control one, given they are two different services.

Reply:
I split the config manager in ad8e1441. For this early-stage PR this works for now, but we'll need eventually to adapt a few handlers (namely: health, password_action, zookeeper, upgrade) because they use a dependency on the charm to access the config manager